protocol/lavasession/single_provider_session.go

```diff
@@ -184,6 +219,22 @@ func (sps *SingleProviderSession) PrepareSessionForUsage(ctx context.Context, cu
 	return nil
 }
 
+func (sps *SingleProviderSession) validateAndAddBadgeUsedCU(currentCU uint64, maxCu uint64, badgeUser string) error {
+	for {
+		badgeUsedCu := sps.userSessionsParent.atomicReadBadgeUsedComputeUnits(badgeUser)
```

Comment:
so instead of badgeuser here it will be the pointer for &pswc.badgeEpochData[badgeUser]

Reply:
All atomic read/write fixed here: https://github.com/lavanet/lava/commit/a465af37e4b2833ab1e2887e0f0c8a642076573e